node/lib/packaging/PackageInfo.ts

```diff
@@ -0,0 +1,173 @@
+        if (typeof interfacesJson === "object") {
+            Object.keys(interfacesJson).forEach((moduleName: string) => {
+                interfaces.push({
+                    description: interfacesJson[moduleName].description,
+                    name: moduleName,
+                });
+            });
+        }
+
+        let translators: PackageTranslatorInfo[] = [];
+        let translatorsJson: {[moduleName: string]: any} = packageJson.opent2t.translators;
+        if (typeof translatorsJson === "object") {
+            Object.keys(translatorsJson).forEach((moduleName: string) => {
+
+                let interfaceReferences: string[] = translatorsJson[moduleName].interfaces;
+                if (!Array.isArray(interfaceReferences)) {
+                    throw new Error("Missing or invalid translator interface references: " +
+                            JSON.stringify(interfaceReferences));
+                }
+
+                interfaceReferences = interfaceReferences.map(<(value: string) => string>
+                    PackageInfo.resolveModuleName.bind(null, packageJson.name));
+
+                let onboardingReference: string = translatorsJson[moduleName].onboarding;
+                if (!onboardingReference || typeof onboardingReference !== "string") {
+                    throw new Error("Missing or invalid translator onboarding reference: " +
+                            JSON.stringify(onboardingReference));
+                }
+
+                onboardingReference =
+                    PackageInfo.resolveModuleName(packageJson.name, onboardingReference);
+
+                let onboardingProperties: { [propertyName: string]: string } =
+                        translatorsJson[moduleName].onboardingProperties;
+                if (!onboardingProperties || typeof onboardingProperties !== "object") {
+                    throw new Error("Missing or invalid translator onboarding properties: " +
+                            JSON.stringify(onboardingProperties));
+                }
+
+                translators.push({
+                    description: translatorsJson[moduleName].description,
+                    interfaces: interfaceReferences,
+                    name: moduleName,
+                    onboarding: onboardingReference,
+                    onboardingProperties: onboardingProperties,
+                });
+            });
+        }
+
+        return {
+            description: packageJson.description,
+            interfaces: interfaces,
+            name: packageJson.name,
+            translators: translators,
+            version: packageJson.version,
+        };
+    }
+
+    /**
+     * Converts references to modules within the same package (using a ./ prefix) into
+     * package-qualified module references.
+     */
+    private static resolveModuleName(packageName: string, moduleName: string): string {
+        return moduleName.startsWith("./") ? (packageName + "/" + moduleName.substr(2)) : moduleName;
```

Comment:
Sure, there is `using("path").join(a, b)`. However that will join using an OS-dependent separator. But here we actually just want a forward-slash, because that's what module paths always use.

---
In reply to: [73275164](https://github.com/openT2T/opent2t/pull/2#discussion_r73275164) [](ancestors = 73275164)